cmd/kube-proxy/app/server.go

```diff
@@ -751,14 +751,14 @@ func (s *ProxyServer) Run() error {
 	serviceConfig.RegisterEventHandler(s.Proxier)
 	go serviceConfig.Run(wait.NeverStop)
 
-	if s.UseEndpointSlices {
+	if endpointsHandler, ok := s.Proxier.(config.EndpointsHandler); ok && s.UseEndpointSlices {
```

Comment:
Is  if s.UseEndpointSlices or ! s.UseEndpointSlices

Reply:
doh!